main.go

```diff
@@ -31,19 +31,26 @@ func init() {
 	}
 }
 
-func main() {
-	err := app.Flag.Parse(os.Args[1:])
+func runArgs(argsStr []string) error {
+	err := app.Flag.Parse(argsStr)
 	if err != nil {
 		log.Printf("error parsing flags: %v\n", err)
-		os.Exit(1)
+		return err
```

Comment:
as `run` returns an `error` now, I would actually remove the `log.Printf` above and return the "decorated" error like so:

```go
return fmt.Errorf("could not parse flags: %v", err)
```

Reply:
Done.